src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSerializationVisitor.cs

```diff
@@ -180,18 +180,18 @@ internal async Task VisitEdmSchemaAsync(EdmSchema element, IEnumerable<KeyValueP
 
             await this.schemaWriter.WriteSchemaElementHeaderAsync(element, alias, mappings).ConfigureAwait(false);
 
-            VisitSchemaElements(element.SchemaElements);
+            await VisitSchemaElementsAsync(element.SchemaElements).ConfigureAwait(false);
 
             // process the functions/actions seperately
             foreach (var operation in element.SchemaOperations)
             {
                 await this.schemaWriter.WriteSchemaOperationsHeaderAsync(operation).ConfigureAwait(false);
-                VisitSchemaElements(operation.Value.AsEnumerable<IEdmSchemaElement>()); // Call AsEnumerable() to make .net 3.5 happy
+                await VisitSchemaElementsAsync(operation.Value.AsEnumerable<IEdmSchemaElement>()).ConfigureAwait(false); // Call AsEnumerable() to make .net 3.5 happy

```

Comment:
Do we still need code to make with `.net 3.5` happy?

Reply:
@habbes I will remove that comment. I think it is already added if you look closely at the `operation.Value.AsEnumerable<IEdmSchemaElement>()` passed as argument to that method.